tokio/src/net/tcp/socket.rs

```diff
@@ -482,6 +482,47 @@ impl TcpSocket {
         let mio = self.inner.listen(backlog)?;
         TcpListener::new(mio)
     }
+
+    /// Converts a `socket2::Socket` into a `TcpSocket`. The socket2 socket
+    /// needs to be in a disconnected state when passed as an argument.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use tokio::net::TcpSocket;
+    /// use socket2::{Domain, Socket, Type};
+    /// use std::net::TcpStream;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
```

Comment:
Your example doesn't compile.
```suggestion
    /// async fn main() -> std::io::Result<()> {
```